lib/live_fixtures/import/insertion_order_computer.rb

```diff
@@ -0,0 +1,127 @@
+class LiveFixtures::Import
+  # :nodoc:
+  class InsertionOrderComputer
+    # :nodoc:
+    class Node
+      attr_reader :path
+      attr_reader :class_name
+      attr_reader :klass
+
+      # The classes this node depends on
+      attr_reader :dependencies
+
+      def initialize(path, class_name, klass)
+        @path = path
+        @class_name = class_name
+        @klass = klass
+        @dependencies = Set.new
+      end
+    end
+
+    def self.compute(table_names, class_names = {}, polymorphic_associations = {})
+      new(table_names, class_names, polymorphic_associations).compute
+    end
+
+    def initialize(table_names, class_names = {}, polymorphic_associations = {})
+      @table_names = table_names
+      @class_names = class_names
+      @polymorphic_associations = polymorphic_associations
+    end
+
+    def compute
+      nodes = build_nodes
+      compute_insert_order(nodes)
+    end
+
+    private
+
+    # Builds an Array of Nodes, each containing dependencies to other nodes
+    # using their class names.
+    def build_nodes
+      # Create a Hash[Class => Node] for each table/class
+      nodes = Hash[@table_names.map do |path|
```

Comment:
oh, i didn't know that Hash[[k,v]] was a way of constructing a hash!

🎨 that being said, this constructor is a little hard to read 😆

Reply:
Yeah, I agree it's a bit hard to read because it's doing too many things at once, plus there's a lot of indent.

I changed it to a simple `nodes = {}` and fill it by iterating `@table_names`. I think it looks simpler now.